carmm/run/workflows/react.py

```diff
@@ -579,32 +580,34 @@ def _calc_generator(params,
     from carmm.run.aims_calculator import get_aims_and_sockets_calculator
     """On machines where ASE and FHI-aims are run separately (e.g. ASE on login node, FHI-aims on compute nodes)
     we need to specifically state what the name of the login node is so the two packages can communicate"""
-    sockets_calc, fhi_calc = get_aims_and_sockets_calculator(dimensions=dimensions,
-                                                             logfile=f"{directory}/socketio.log",
-                                                             verbose=True,
-                                                             codata_warning=False,
-                                                             directory=directory
-                                                             )
 
     """Remove previous xc argument to ensure libxc warning override is first"""
-    fhi_calc.parameters = {k: v for k, v in fhi_calc.parameters.items() if k != 'xc'}
-    fhi_calc.parameters['override_warning_libxc'] ='True'
+    xc = params['xc']
+    fhi_params = {k: v for k, v in params.items() if k != 'xc'}
+    fhi_params['override_warning_libxc'] = 'True'
+    fhi_params['xc'] = xc
 
     """Forces required for optimisation"""
     if not forces:
-        fhi_calc.parameters = {k: v for k, v in fhi_calc.parameters.items() if k != 'compute_forces'}
+        fhi_params = {k: v for k, v in fhi_params.items() if k != 'compute_forces'}
 
     """Add analytical stress keyword for unit cell relaxation"""
     if relax_unit_cell:
         assert dimensions == 3, "Strain Filter calculation requested, but the system is not periodic in 3 dimensions."
-        fhi_calc.parameters['compute_analytical_stress'] = 'True'
+        fhi_params['compute_analytical_stress'] = 'True'
 
-    """Set a unique .out output name"""
-    fhi_calc.outfilename = out_fn
+    sockets_calc, fhi_calc = get_aims_and_sockets_calculator(dimensions=dimensions,
+                                                             logfile=f"{directory}/socketio.log",
+                                                             verbose=True,
+                                                             codata_warning=False,
+                                                             directory=directory,
+                                                             **fhi_params)
 
-    """FHI-aims settings set up"""
-    for k, v in params.items():
-        fhi_calc.parameters[k] = v
```

Comment:
Hi @ikowalec, I am not sure what line 606-607 meant to do. I thought this is to add the `'xc'` back to `fhi_calc.parameters`, as `'xc'` is removed at the start. But this would also add other parameters from `params`.

Say we removed `'compute_forces'`, these two lines would restore it.